includes/class-wcpdf-main.php

```diff
@@ -846,36 +850,78 @@ public function set_header_logo_height( $document_type, $document = null ) {
+		$delete_timestamp = time() - ( intval ( DAY_IN_SECONDS * $cleanup_age_days ) );
+		$this->temporary_files_cleanup( $delete_timestamp );
+	}
+	
 	/**
-	 * Remove attachments older than 1 week (daily, hooked into wp_scheduled_delete )
+	 * Remove temporary files from paths older than 1 week (daily, hooked into wp_scheduled_delete )
+	 * @param string|int  $delete_timestamp timestamp
+	 * 
+	 * @return array Output message
 	 */
-	public function attachments_cleanup() {
-		if ( !function_exists("glob") || !function_exists('filemtime') ) {
-			// glob is required
-			return;
-		}
+	public function temporary_files_cleanup( $delete_timestamp = '' ) {
+		global $wp_filesystem;
 
-		
-		if ( !isset( WPO_WCPDF()->settings->debug_settings['enable_cleanup'] ) ) {
-			return;
+		if ( ! function_exists( 'WP_Filesystem' ) ) {
+			require_once( ABSPATH . 'wp-admin/includes/file.php' );
 		}
 
+		if ( ! WP_Filesystem() || is_null( WP_Filesystem() ) ) {
+			/* translators: WP_Filesystem */
+			return array( 'error' => sprintf( esc_html__( "%s couldn't be initiated!", 'woocommerce-pdf-invoices-packing-slips' ), 'WP_Filesystem' ) );
+		}
 
-		$cleanup_age_days = isset(WPO_WCPDF()->settings->debug_settings['cleanup_days']) ? floatval(WPO_WCPDF()->settings->debug_settings['cleanup_days']) : 7.0;
-		$delete_timestamp = time() - ( intval ( DAY_IN_SECONDS * $cleanup_age_days ) );
-
-		$tmp_path = $this->get_tmp_path('attachments');
+		$paths_to_cleanup    = apply_filters( 'wpo_wcpdf_cleanup_tmp_paths', array(
+			$this->get_tmp_path( 'attachments' ),
+			$this->get_tmp_path( 'dompdf' ),
+		) );
+		$exclude_directories = apply_filters( 'wpo_wcpdf_cleanup_exclude_directories', array() );
+		$allowed_filetypes   = apply_filters( 'wpo_wcpdf_cleanup_allowed_filetypes', array( 'pdf' ) );
+		$folders_level       = apply_filters( 'wpo_wcpdf_cleanup_folders_level', 3 );
+		$files               = array();
+		$success             = 0;
+		$error               = 0;
+		$output              = array();
+
+		foreach ( $paths_to_cleanup as $path ) {
+			$files = array_merge( $files, list_files( $path, $folders_level, $exclude_directories ) );
+		}
 
-		if ( $files = glob( $tmp_path.'*.pdf' ) ) { // get all pdf files
-			foreach( $files as $file ) {
-				if( is_file( $file ) ) {
-					$file_timestamp = filemtime( $file );
-					if ( !empty( $file_timestamp ) && $file_timestamp < $delete_timestamp ) {
-						@unlink($file);
+		if ( ! empty( $files ) ) {
+			foreach ( $files as $file ) {
+				$filetype = wp_check_filetype( $file ); // 'php' files return 'false' for 'ext'
+				if ( $filetype['ext'] != false && in_array( $filetype['ext'], $allowed_filetypes ) && $wp_filesystem->exists( $file ) ) {
+					$file_timestamp  = $wp_filesystem->mtime( $file );
+					$can_delete_file = ! empty( $delete_timestamp ) && is_numeric( $delete_timestamp ) ? $file_timestamp < intval( $delete_timestamp ) : true;
```

Comment:
sanitization of `$delete_timestamp` should be done at the beginning of this method, this makes the code a lot more readable, because you can then just do:
```php
if ( $file_timestamp < $delete_before ) {
```